kvui.py

```diff
@@ -3,6 +3,7 @@
 import sys
 import typing
 import re
+from typing import *
```

Comment:
Please don't use `*` imports.

It makes it harder for readers to know where things come from.
It makes it harder for tools to check for undefined names and unused imports.
It makes debugging more difficult if someone accidentally uses a name that they didn't know was imported into the scope.

Reply:
Since the original is using syntax like `typing.List`, I recommend you follow their lead here.

Reply:
It pollutes the global namespace,  though for modern python the effect is certainly less than it used to be.

Reply:
`typing` is both the only `*` import I ever use myself and something I almost always `*` import, as it should just be part of the base language, frankly. If core devs don't like it I'll change it, though I'd recommend people try it out in their own code sometimes, it's been a real QoL improvement especially for <=3.9 code.

Reply:
Updated, only importing `List`, `Tuple`, `Optional`. Tested it by opening the client and doing a couple of commands.

Reply:
It still has
```
import typing
from typing import List, Tuple, Optional
```

feels very weird having both, imo

Reply:
I think having both can be ok if one is meant for < 3.10 stuff and the other for stuff that would not be needed after dropping support for the earlier version.

Reply:
Having both seems fairly common. BaseClasses.py, settings.py, Zelda1Client.py, Utils.py all have this (though the from import is unused in Zelda1Client.py).

I don't see this as particularly related to this PR and frankly there doesn't seem to be a single consist style. I'm personally happy to leave it, and won't touch this until a reviewer says which I should prefer.

As I said above, I am a `from typing import *` person, so both these options seem equally inconvenient to me. `import typing` adds seven characters per type annotation to a line, and as 3~4 type annotations on a line is fairly common, I find this to be very unreadable or requires many extra line-breaks. `from typing import <items>` requires going back and updating the import every time a new type is used in an annotation. If people really dislike typing star imports, the best I could recommend is having a style guide that encourages always importing a baseline of common-use names, e.g. `from typing import List, Tuple, Dict, Set, Union, Optional, Any`.

Reply:
There is a style-guide, see style.md in the docs, which says to follow PEP8, which says to avoid using `from <module> import *`

The style guide also says "Worlds that do not follow PEP8 should still have a consistent style across its files to make reading easier."

Ideally, someone would go back and clean up those few things in those files, but it's been over a year for Utils at least so I don't see anyone doing that. The best we can do is try to be good about following it going forward.

Reply:
I'll also note the `typing` module defines a well-maintained `__all__` to control what gets imported by `import *`, and it's been present since at least Python 3.7 (oldest I have installed on my machine to check). If the mechanistic argument doesn't do it for you, I've been doing `from typing import *` in almost all my Python scripts for at least 3 years, debugging often, and have never had an issue come about from this.

Reply:
> see style.md in the docs

Thanks for the link, though it's still insufficient. Did the best I could, because I kind of just want this conversation over.

Decisions:
* To avoid the aforementioned double-import, I went with what was there before (`import typing`), and updated my type annotations to that style.
* This pushed the `def keyboard_on_key_down()` line to over 120 characters, so I had to break it up. [PEP8 is silent on how muilti-line parameter lists should work with type annotations, and says it's up to the user/team](https://github.com/python/typing/issues/433#issuecomment-302522408). I've used the style I prefer, which is also in that thread.
  * style.md makes no comment on how to do multi-line parameter lists with type annotations.
* I've also corrected the `self._command_history = ` line just above it as it was 124 characters wide.

I hope this is enough to end the conversation `:P`

Reply:
It probably didn't need to go as long as it did, agreed